recipe/meta.yaml

```diff
@@ -14,156 +15,228 @@
+        - cuda-nvtx-dev
+        - cuda-profiler-api
+        - cuda-cccl_{{ target_platform }}
+        - libcublas-dev
+        - libcufft-dev
+        - libcurand-dev
+        - libcusolver-dev
+        - libcusparse-dev
+        {% endif %}
+        # optional dependencies
+        # TODO: see https://github.com/conda-forge/cudnn-feedstock/issues/58
+        - cudnn ~=8.8           # [not (aarch64 or (ppc64le and (cuda_compiler_version or "").startswith("12")))]
+        - nccl ~=2.16           # [linux]
+        - cutensor ~=2.0
+        - cusparselt ~=0.2.0    # [(linux64 or aarch64 or win) and (cuda_compiler_version or "").startswith("11")]
+      run:
+        - python
+        - {{ pin_compatible('fastrlock', max_pin='x.x') }}
+        - numpy ~=1.22
+        - cuda-version >={{ cuda_major }}.2,<{{ cuda_major+1 }}  # [(cuda_compiler_version or "").startswith("11")]
+        - cuda-version >={{ cuda_major }}.0,<{{ cuda_major+1 }}  # [(cuda_compiler_version or "").startswith("12")]
+      run_constrained:
+        # Only GLIBC_2.17 or older symbols present
+        - __glibc >=2.17  # [linux]
+        - cudatoolkit                                            # [(cuda_compiler_version or "").startswith("11")]
+        - cuda-nvrtc                                             # [(cuda_compiler_version or "").startswith("12")]
+        - cuda-nvtx                                              # [(cuda_compiler_version or "").startswith("12")]
+        - libcublas                                              # [(cuda_compiler_version or "").startswith("12")]
+        - libcufft                                               # [(cuda_compiler_version or "").startswith("12")]
+        - libcurand                                              # [(cuda_compiler_version or "").startswith("12")]
+        - libcusolver                                            # [(cuda_compiler_version or "").startswith("12")]
+        - libcusparse                                            # [(cuda_compiler_version or "").startswith("12")]
+        - scipy ~=1.7
+        - optuna ~=3.0
+        - __cuda >=11.2                                          # [(cuda_compiler_version or "").startswith("11")]
+        - __cuda >=12.0                                          # [(cuda_compiler_version or "").startswith("12")]
+        # TODO: see https://github.com/conda-forge/cudnn-feedstock/issues/58
+        - {{ pin_compatible('cudnn') }}                          # [not (aarch64 or (ppc64le and (cuda_compiler_version or "").startswith("12")))]
+        - {{ pin_compatible('nccl') }}                           # [linux]
+        - {{ pin_compatible('cutensor') }}
+        - {{ pin_compatible('cusparselt', max_pin='x.x') }}      # [(linux64 or aarch64 or win) and (cuda_compiler_version or "").startswith("11")]
+    test:
+      requires:
+        # need the libcuda stub for import test
+        - cuda-driver-dev  # [linux and (cuda_compiler_version or "").startswith("12")]
```

Comment:
Think we should add these as test requirements to make sure these components are covered when running the test suite (offline)

```suggestion
        - cuda-driver-dev  # [linux and (cuda_compiler_version or "").startswith("12")]
        - cudatoolkit      # [(cuda_compiler_version or "").startswith("11")]
        - cuda-nvrtc       # [(cuda_compiler_version or "").startswith("12")]
        - cuda-nvtx        # [(cuda_compiler_version or "").startswith("12")]
        - libcublas        # [(cuda_compiler_version or "").startswith("12")]
        - libcufft         # [(cuda_compiler_version or "").startswith("12")]
        - libcurand        # [(cuda_compiler_version or "").startswith("12")]
        - libcusolver      # [(cuda_compiler_version or "").startswith("12")]
        - libcusparse      # [(cuda_compiler_version or "").startswith("12")]
        - scipy ~=1.7
        - optuna ~=3.0
        # TODO: see https://github.com/conda-forge/cudnn-feedstock/issues/58
        - cudnn            # [not (aarch64 or (ppc64le and (cuda_compiler_version or "").startswith("12")))]
        - nccl             # [linux]
        - cutensor
        - cusparselt       # [(linux64 or aarch64 or win) and (cuda_compiler_version or "").startswith("11")]
```

Reply:
I think I messed up. Let me fix it. The existing tests should go with `cupy` not `cupy-core`.

Reply:
done in 1dc6b5e

Reply:
Ah ok. Just want to make sure that we have a batteries included test somewhere

Reply:
Yup that was the intent! `cupy` should continue to be fully functional as before.